### PR TITLE
feat(tiflow): support downloading dependencies from a specified branch

### PR DESCRIPTION
**Pull request by purelind**
support downloading dependencies from a specified branch in PR titles.

Related to this PR: https://github.com/pingcap/tiflow/pull/11572

### CODE REVIEW

File: pipelines/pingcap/tiflow/latest/pull_cdc_integration_kafka_test.groovy [hunk 1]
```diff
@@ -84,17 +84,27 @@ pipeline {
             steps {
                 dir("third_party_download") {
                     retry(2) {
```
Comment (wuhuizuo) [inline pipelines/pingcap/tiflow/latest/pull_cdc_integration_kafka_test.groovy:86]: just move download part into the retry closure。

**Comment by purelind** [comment]
replay test link
- [ ] cdc kafka https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftiflow%2Fpull_cdc_integration_kafka_test/detail/pull_cdc_integration_kafka_test/2715/pipeline
- [ ] cdc mysql  https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftiflow%2Fpull_cdc_integration_test/detail/pull_cdc_integration_test/2638/pipeline
- [ ] cdc pulsar https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftiflow%2Fpull_cdc_integration_pulsar_test/detail/pull_cdc_integration_pulsar_test/2316/pipeline

**Review by wuhuizuo — APPROVED** [review]

**Pull request merged**